marathon_acme/sse_protocol.py

```diff
@@ -181,11 +183,32 @@ def _prepare_data(self):
         # Add a newline character between lines
         return '\n'.join(self._data_lines)
 
-    def connectionLost(self, reason=connectionDone):
-        self.log.failure('SSE connection lost', reason, LogLevel.warn)
+    def connectionLost(self, reason):
         self.setTimeout(None)  # Cancel the timeout
-        for d in list(self._waiting):
-            d.callback(None)
+
+        # Inspired by treq:
+        # https://github.com/twisted/treq/blob/release-17.8.0/src/treq/content.py#L34-L41
+        if reason.check(ResponseDone):
+            self.log.info('SSE connection ended')
+            self._callback_waiting(None)
+        elif reason.check(PotentialDataLoss):
+            # PotentialDataLoss errors are not very interesting, apparently,
+            # and they clog up the logs. Ignoring them seems common in Twisted.
+            # http://twistedmatrix.com/trac/ticket/4840
```

Comment:
It basically means "the connection ended and we didn't know that we got all the data" -- typically because there's no `Content-length` header because we're streaming an indeterminate amount of data.